src/realm/transaction.cpp

```diff
@@ -45,32 +44,71 @@ ColInfo get_col_info(const Table* table)
     return cols;
 }
 
+void add_list_to_repl(CollectionBase& list, Replication& repl, util::UniqueFunction<void(Mixed)>& update_embedded);
+void add_dictionary_to_repl(Dictionary& dict, Replication& repl, util::UniqueFunction<void(Mixed)>& update_embedded)
```

Comment:
I would prefer that `update_embedded` is transfered by value. It is just a pointer.